kpi/deployment_backends/kc_reader/utils.py

```diff
@@ -0,0 +1,68 @@
+from django.contrib.auth.models import User
+from django.db import models
+from django.db import ProgrammingError
+
+
+class __ReadOnlyModelError(ValueError):
+    pass
+
+
+class __ReadOnlyModel:
```

Comment:
Yeah. I think making the models private_ish_ is the main safeguard. Overriding the methods is secondary.